### ISSUE TITLE
[8.7.1] GRU gradient truncation에 대한 이해

### ISSUE THREAD

**Issue by jiwon-ryu** [woowacampers/d2l-pytorch#5]
8.7.1.1 - 8.7.1.4에서는 다양한 gradient truncation 방법들이 소개되어 있는데요, 아래의 Eq. 8.7.7로부터 발생될 수 있는 gradient vanishing/exploding 방지를 위한 목적인 것 같습니다.

![image](https://user-images.githubusercontent.com/69613571/108468064-82992680-72c9-11eb-8cf0-a1b2444c54e6.png)

그리고 Fig. 8.7.1과 같은 그림이 제시되어 있었는데요,

![image](https://user-images.githubusercontent.com/69613571/108468193-afe5d480-72c9-11eb-8292-ae6426c19485.png)

그렇다면 truncation이라는 것이 모든 시간에 대해 gradient를 구하되, 전 구간을 한꺼번에 커버하는 것이 아니라 여러번 쪼개서 계산한다음 합치는 것으로 이해하면 될까요?

**Comment by dhh0** [comment]
네, 저도 마찬가지로 원래 BPTT는 맨 처음 time step까지 보고 truncation은 나누어진 구간까지만 gradient를 계산하는 것으로 이해했습니다.

**Comment by staycozyboy** [comment]
저도 생각해보다가 깔끔하게 이해되지 않는 부분이 있는데, 그림을 보면

avg(sum(t_n step 부터 t_n-k step까지 back prop) + sum(**t_n-k-1 step 부터** t_n-2k-1 step까지 back prop) + ...)

을 묘사하고 있는 것으로 이해가 됩니다. <br>
근데, 제가 이해하는 바로는

avg(sum(t_n step 부터 t_n-k step까지 back prop) + sum(**t_n-1 step 부터** t_n-k-1 step까지 back prop) + ...)

이 되어야 하는 것 같은데, 어떤 점이 틀린 지 설명 부탁드립니다 ㅠㅠ
